server/Mistakes.Journal.Api/Api/Mistakes/Controllers/MistakesController.cs

```diff
@@ -46,22 +68,65 @@ public async Task<ActionResult<MistakeWebModel>> CloneMistake(Guid mistakeId)
         {
             var mistake = await _dataContext.Set<Mistake>()
                 .Include(m => m.Tips)
+                .Include(m => m.MistakeLabels)
+                .ThenInclude(m => m.Label)
+                .Include(m => m.Repetitions)
                 .SingleOrDefaultAsync(m => m.Id == mistakeId);
 
             if (mistake is null)
                 return NotFound();
 
-            var newMistake = new NewMistakeWebModel
-            {
-                Name = mistake.Name,
-                Goal = mistake.Goal,
-                Priority = mistake.Priority,
-                Tips = mistake.Tips.Select(t => t.Content).ToList(),
-            };
+            var newMistake = mistake.ToNewMistakeWebModel();
 
             return await AddMistake(newMistake);
         }
 
+        [HttpPost("{mistakeId:guid}/add-repetiton")]
+        public async Task<ActionResult<MistakeWebModel>> AddRepetitionResetCounter(Guid mistakeId, [FromBody, MJIncorrectMistakeDate(1)] DateTime? dateTime)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            var mistake = await _dataContext.Set<Mistake>()
+                .Include(m => m.Repetitions)
+                .SingleOrDefaultAsync(m => m.Id == mistakeId);
+
+            if (mistake is null)
+                return NotFound();
+
+            mistake.Repetitions.Add(new Repetition(dateTime ?? DateTime.Now));
+
+            _dataContext.Entry(mistake).State = EntityState.Modified;
```

Comment:
Again, bad practice :/ Why do you need to use it this way? It should by default see the state change. Maybe try adding repetition to Set<Repetitions> and to repetition assign mistake? I don't know, just a idea